src/coredump/coredump.c

```diff
@@ -1462,6 +1478,33 @@ static int process_kernel(int argc, char* argv[]) {
                 log_open();
         }
 
+        r = in_same_namespace(getpid_cached(), context.pid, NAMESPACE_PID);
+        if (r < 0)
+                log_debug_errno(r, "Failed to check pidns of crashing process, ignoring: %m");
+
+        if (r == 0 && getenv_bool("SYSTEMD_COREDUMP_ALLOW_NAMESPACE_CHANGE") > 0) {
+                int fd;
+
+                r = namespace_open(context.pid,  NULL, &fd, NULL, NULL, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open mntns of crashing process: %m");
+
+                mntns_fd = fd;
+        } else {
+                /* Crashing process is not running in the container or changing namespace is disabled, but we
+                   still need to send mount namespace fd along side coredump fd so let's just open our own
+                   mount namespace. Entering it will be NOP but that is OK. */
```

Comment:
Why? The ELF parsing code seems to handle `mntns_fd` being unset.

Reply:
I wanted to keep special casing limited to this only. So the code which is sending fds would work the same way in both cases.

Reply:
But if that is a blocker for you I can change it.

Reply:
Not a blocker. It just confused me, as I'd expected this would only be needed for the special case.